examples/vae_simple.py

```diff
@@ -0,0 +1,322 @@
+            # will cause the loss function to become NAN. So we set the limit
+            # stdev >= exp(-1 * relu_shift)
+            relu_shift = 10
+            l_dec_logsigma = nn.layers.DenseLayer(l_dec_hid, num_units=x_dim,
+                    W = nn.init.GlorotUniform() if W_dec_ls is None else W_dec_ls,
+                    b = nn.init.Constant(0) if b_dec_ls is None else b_dec_ls,
+                    nonlinearity = lambda a: T.nnet.relu(a+relu_shift)-relu_shift,
+                    name='dec_logsigma')
+            l_output = GaussianSampleLayer(l_dec_mu, l_dec_logsigma,
+                    name='dec_output')
+            l_dec_mu_list.append(l_dec_mu)
+            l_dec_logsigma_list.append(l_dec_logsigma)
+            l_output_list.append(l_output)
+            if W_dec_hid is None:
+                W_dec_hid = l_dec_hid.W
+                b_dec_hid = l_dec_hid.b
+                W_dec_mu = l_dec_mu.W
+                b_dec_mu = l_dec_mu.b
+                W_dec_ls = l_dec_logsigma.W
+                b_dec_ls = l_dec_logsigma.b
+        l_Z_list.append(l_Z)
+    l_output = nn.layers.ElemwiseSumLayer(l_output_list, coeffs=1./L, name='output')
+    return l_enc_mu, l_enc_logsigma, l_dec_mu_list, l_dec_logsigma_list, l_output_list, l_output
+
+# ############################## Main program ################################
+
+# helper function for log-likelihood expression
+def log_likelihood(tgt, mu, ls):
+    return T.sum(-(np.float32(0.5 * np.log(2 * np.pi)) + ls)
+            - 0.5 * T.sqr(tgt - mu) / T.exp(2 * ls))
+
+def main(L=2, z_dim=2, n_hid=256, num_epochs=20, binary=True):
+    print("Loading data...")
+    X_train, X_val, X_test = load_dataset()
+    X_train_tgt = X_train.reshape(X_train.shape[0],-1)
+    X_val_tgt = X_val.reshape(X_val.shape[0],-1)
+    X_test_tgt = X_test.reshape(X_test.shape[0],-1)
+    width, height = X_train.shape[2], X_train.shape[3]
+    input_var = T.tensor4('inputs')
+    target_var = T.dmatrix('targets')
+
+    # Create VAE model
+    print("Building model and compiling functions...")
+    print("L = {}, z_dim = {}, n_hid = {}, binary={}".format(L, z_dim, n_hid, binary))
+    x_dim = width * height
+    l_z_mu, l_z_ls, l_x_mu_list, l_x_ls_list, l_x_list, l_x = \
+           build_vae(input_var, L=L, binary=binary, z_dim=z_dim, n_hid=n_hid)
+
+    # If there are dropout layers etc these functions return masked or non-masked expressions
+    # depending on if they will be used for training or validation/test err calcs
+    z_mu = lambda b: nn.layers.get_output(l_z_mu, deterministic=b)
+    z_ls = lambda b: nn.layers.get_output(l_z_ls, deterministic=b)
+    x_mu = lambda b: [nn.layers.get_output(l_x_mu, deterministic=b) for l_x_mu in l_x_mu_list]
+    x_ls = lambda b: [nn.layers.get_output(l_x_ls, deterministic=b) for l_x_ls in l_x_ls_list]
+    x_list = lambda b: [nn.layers.get_output(l_x1, deterministic=b) for l_x1 in l_x_list]
```

Comment:
You should use a single `get_output()` call for all of that, unless you are perfectly sure that none of the outputs overlap in the noise layers used (where noise layers include dropout layers and your gaussian sampling layer). Otherwise the different outputs will use separate forward passes with separately sampled noise!